Homework/src/main/java/io/codelex/javaadvancedtest/exercise1/CreditCard.java

```diff
@@ -0,0 +1,30 @@
+package io.codelex.javaadvancedtest.exercise1;
+
+public class CreditCard extends Card {
+
+    public CreditCard(int number, String owner, int ccv, double balance) {
+        super(number, owner, ccv, balance);
+    }
+
+    @Override
+    public void addMoney(double amount) {
+        super.balance += amount;
+    }
+
+    @Override
+    public void takeMoney(double amount) throws NotEnoughFundsException {
+        if (balance >= amount) {
+            balance -= amount;
+        } else {
+            throw new NotEnoughFundsException("Not enough funds");
+        }
+        if (balance > 100){
```

Comment:
Līdz šai vietai takeMoney sakrīt ar DebitCard takeMoney. Attiecīgi to kopīgo pusi varētu pārnest uz abstrakto Card klasi, tādā veidā šeit paliktu:
```
super.takeMoney(amount);
if (balance > 100){
   displayWarning();
}
```
Savukārt DebitCard varētu ņemt tad ārā to takeMoney jo viņa strādā 1 pret 1 kā abstraktā klase.